### PR TITLE
fix: Expand allowed characters in keys #4044

### PR DESCRIPTION
**Pull request by uniqueeest**
### Changes
This modification extends the definition of the key to include spaces, colon, slash, hyphen, and parentheses. It also changes the key to include more character types while maintaining the pattern that ends in colon.

### Checklist
- [ ] Added markup tests, or they don't apply here because...
- [ ] Updated the changelog at `CHANGES.md`

### CODE REVIEW

File: src/languages/yaml.js [hunk 1]
```diff
@@ -20,12 +20,12 @@ export default function(hljs) {
...
     className: 'attr',
     variants: [
-      // added brackets support 
-      { begin: /\w[\w :()\./-]*:(?=[ \t]|$)/ },
-      { // double quoted keys - with brackets
-        begin: /"\w[\w :()\./-]*":(?=[ \t]|$)/ },
-      { // single quoted keys - with brackets
-        begin: /'\w[\w :()\./-]*':(?=[ \t]|$)/ },
+      // added brackets and other special characters support
+      { begin: /\w[\w\s:/.\-()]*:(?=[ \t]|$)/ },
```
Comment (joshgoebel) [inline src/languages/yaml.js:24]: You just changed the order - other than `' '` and `\s`... why?

And is \s REALLY what we want?  That can include tabs and carriage returns... is there a reason a literal space we not sufficient?

Comment (uniqueeest) [reply src/languages/yaml.js]: @joshgoebel I think I got it wrong. I changed it to just add * and @, what do you think?

File: src/languages/yaml.js [hunk 2]
```diff
@@ -20,12 +20,12 @@ export default function(hljs) {
...
-      // added brackets support 
-      { begin: /\w[\w :()\./-]*:(?=[ \t]|$)/ },
+      // added brackets support
+      { begin: /\w[\w :()\./\-*@]*:(?=[ \t]|$)/ },
       { // double quoted keys - with brackets
-        begin: /"\w[\w :()\./-]*":(?=[ \t]|$)/ },
+        begin: /"\w[\w :()\./\-*@]*":(?=[ \t]|$)/ },
       { // single quoted keys - with brackets
-        begin: /'\w[\w :()\./-]*':(?=[ \t]|$)/ },
+        begin: /'\w[\w :()\./\-*@]*':(?=[ \t]|$)/ },
```
Comment (joshgoebel) [inline src/languages/yaml.js:28]: Can we please keep the `-` at the very end of the character group so we don't need the extra escape?

Also, could we get a test case added to the markup tests to show that this works as intended?

Comment (uniqueeest) [reply src/languages/yaml.js]: hello! @joshgoebel. Can you please review this part?

**Comment by joshgoebel** [comment]
How are you generating the expect file?  All literal quotes need to be HTML escaped.

**Comment by uniqueeest** [comment]
@joshgoebel I wrote it.

expect-test
```
<span class="hljs-attr">"*":</span>
```
test
```
"*":
```
But it comes out like the picture I posted.

**Comment by joshgoebel** [comment]
Every `"` should be `&quot;` as the failing test is telling you.  The output from Highlight.js must be HTML encoded.

**Comment by uniqueeest** [comment]
@joshgoebel I only changed the regular expression part from the original code. Should I add encoding in the test code part?

**Comment by uniqueeest** [comment]
@joshgoebel Test file modification completed. It also passed without any problems.

File: test/markup/yaml/complex-keys.txt [hunk 3]
```diff
@@ -1,18 +1,13 @@
-"*":
-  "**/.env"
+"**/.env"
 
-"octo-repo":
```
Comment (joshgoebel) [inline test/markup/yaml/complex-keys.txt]: Why did we lose all the keys?

Comment (uniqueeest) [reply test/markup/yaml/complex-keys.txt]: @joshgoebel It wasn't a key value that needed to be tested, but a title for that test😂. I was mistaken at first and added that part to the test.

Comment (uniqueeest) [reply test/markup/yaml/complex-keys.txt]: @joshgoebel It wasn't a key value that needed to be tested, but a title for that test😂. I was mistaken at first and added that part to the test.

File: src/languages/yaml.js [hunk 4]
```diff
@@ -20,12 +20,12 @@ export default function(hljs) {
   const KEY = {
```
Comment (joshgoebel) [inline src/languages/yaml.js:20]: Should we add a look-ahead for `:` at the end perhaps to make sure these are keys?

Comment (uniqueeest) [reply src/languages/yaml.js:20]: @joshgoebel Should I add it if I think I need it? How should I do it

**Comment by joshgoebel** [comment]
#4105 has solved this slightly more flexibly.  Sorry.

**Pull request closed without merging**